cluniverse/Luniverse.cs

```diff
@@ -0,0 +1,384 @@
+                    {
+                        msg = "Invalid amount";
+                        return false;
+                    }
+
+                    BigInteger gain;
+                    if (!BigInteger.TryParse(gainString, out gain))
+                    {
+                        msg = "Invalid gain";
+                        return false;
+                    }
+
+                    transferAmount = transferAmount + gain;
+                    
+                    if (transferAmount < 0)
+                    {
+                        msg = "Overflow";
+                        return false;
+                    }
+
+                    var txCount = web3.Eth.Transactions.GetTransactionCount.SendRequestAsync(ethSrcAddress).Result;
+                    TransactionSigner signer = new TransactionSigner();
+
+                    HexBigInteger gasPrice;
+
+                    string gasPriceInGweiString = cfg["gasPriceInGwei"];
+                    if (int.TryParse(gasPriceInGweiString, out int gasPriceOverride))
+                    {
+                        gasPrice = new HexBigInteger(Nethereum.Util.UnitConversion.Convert.ToWei(gasPriceOverride, Nethereum.Util.UnitConversion.EthUnit.Gwei));
+                    }
+                    else
+                    {
+                        gasPrice = web3.Eth.GasPrice.SendRequestAsync().Result;
+                    }
+
+                    string to;
+                    string data;
+                    BigInteger amount;
+                    HexBigInteger gasLimit;
+                    if (erc20)
+                    {
+                        string creditcoinContract = cfg["creditcoinContract"];
+                        string creditcoinContractAbi = cfg["creditcoinContractAbi"];
+
+                        to = creditcoinContract;
+
+                        var contract = web3.Eth.GetContract(creditcoinContractAbi, creditcoinContract);
+                        var burn = contract.GetFunction("exchange");
+                        var functionInput = new object[] { transferAmount, txBuilder.getSighash() };
+                        data = burn.GetData(functionInput);
+                        gasLimit = burn.EstimateGasAsync(functionInput).Result;
+                        amount = 0;
+                    }
+                    else
+                    {
+                        gasLimit = web3.Eth.Transactions.EstimateGas.SendRequestAsync(new Nethereum.RPC.Eth.DTOs.CallInput(orderId, ethDstAddress, new Nethereum.Hex.HexTypes.HexBigInteger(transferAmount))).Result;
+
+                        to = ethDstAddress;
+                        data = orderId;
+                        amount = transferAmount;
+                    }
+
+                    string txRaw = signer.SignTransaction(ethereumPrivateKey, to, amount, txCount, gasPrice, gasLimit, data);
+                    payTxId = web3.Eth.Transactions.SendRawTransaction.SendRequestAsync("0x" + txRaw).Result;
```

Comment:
Should use EnsureHexPrefix() instead of adding prefix: https://github.com/Nethereum/Nethereum/blob/master/src/Nethereum.Hex/HexConvertors/Extensions/HexByteConvertorExtensions.cs